File: seed/pydantic-v2/trace/resources/v_2/resources/problem/types/default_provided_file.py
from pydantic import BaseModel
from resources.v_2.resources.problem.types.file_info_v_2 import FileInfoV2
from typing import List
from resources.commons.types.variable_type import VariableType
from dt import datetime
from core.datetime_utils import serialize_datetime
class DefaultProvidedFile(BaseModel):
    file: FileInfoV2
    related_types: List[VariableType] = 
    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/v_3/problem/default_provided_file.rb

module Seed
    module Types
        class DefaultProvidedFile < Internal::Types::Model
            field :file, Seed::v_2::v_3::problem::FileInfoV2, optional: false, nullable: false
            field :related_types, Internal::Types::Array[Seed::commons::VariableType], optional: false, nullable: false
        end
    end
end
